### PR TITLE
feat: Implemented authorization

### PR DESCRIPTION
**Pull request by AndrewKaenko**

### CODE REVIEW

File: src/containers/Layout.tsx [hunk 1]
```diff
@@ -21,20 +33,198 @@ const SearchResult = styled(Search)`
...
+
+const LoginButton = styled(Button)`
+  background-color: #1f1f1f;
+  border: none;
+  &:hover {
+    background-color: #177ddc;
+  }
+`;
+
 const Content = styled(Layout.Content)``;
```
Comment (fominmaksim) [inline src/containers/Layout.tsx]: Looks like useless styled component. Why don't you use it directly?

File: src/containers/Layout.tsx [hunk 2]
```diff
@@ -21,20 +33,198 @@ const SearchResult = styled(Search)`
...
 
-const LayoutWrapper: React.FunctionComponent = ({ children }) => (
-  <Layout>
-    <Sidebar />
+const LogInModal = styled(Modal)``;
```
Comment (fominmaksim) [inline src/containers/Layout.tsx:53]: Looks like useless styled component. Why don't you use it directly?

File: src/containers/Layout.tsx [hunk 3]
```diff
@@ -21,20 +33,198 @@ const SearchResult = styled(Search)`
...
+
+const SignInModal = styled(Modal)``;
```
Comment (fominmaksim) [inline src/containers/Layout.tsx]: Looks like useless styled component. Why don't you use it directly?

**Review by fominmaksim — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -17,6 +18,8 @@
     "antd": "^4.16.13",
     "axios": "^0.24.0",
     "dotenv": "^10.0.0",
+    "firebase": "^9.5.0",
+    "firebaseui": "^6.0.0",
```
Comment (alexandrtovmach) [inline package.json]: do we need this?

**Review by alexandrtovmach — APPROVED** [review]

**Pull request merged**